### ISSUE TITLE
Incorrect(?) output for test data

### ISSUE THREAD

**Issue by Yamahari** [Jokeren/DataMining-gSpan#27]
Hi, 
I was trying out gBolt on some test data I created and I am not sure whether or not I am getting the expected output.
I tested the same data on a gaston miner implementation and provided the seemingly correct output in the following .zip.

[gBolt-Tests.zip](https://github.com/Jokeren/gBolt/files/6913286/gBolt-Tests.zip)

"graphs_numeric.lg" is the input file, "graphs.lg" is the original file with string labels, i also plotted the input and output graphs for visual aid.

Maybe you can take a look 🙂

**Comment by Jokeren** [comment]
Thanks for the data. Do you mean gBolt's result is different from gaston's?

**Comment by Yamahari** [comment]
Yes very different, the output looks really odd, but I am not sure whether or not I am at fault for providing a faulty input/ gSpan is not able to handle this kind of graph/implementation bug
I used the gaston miner implemented here https://github.com/timtadh/parsemis

**Comment by Jokeren** [comment]
It's good to know the other project.

The graph data you attached seems violating gspan/gbolt's specification.

https://github.com/Jokeren/gBolt#input-specification

`<vertex-id>` must be contiguous without gaps

**Comment by Jokeren** [comment]
I'll take a further look once have time. The code itself does not force the constraint and I forgot why I wrote the specification like this.

**Comment by Yamahari** [comment]
So the vertex id should not start at 0 again if there's more than 1 graph?

**Comment by Yamahari** [comment]
I tried with ordered vertices and the output seems to be the same, here's the input file with ordered
[graphs_numeric.txt](https://github.com/Jokeren/gBolt/files/6915836/graphs_numeric.txt)
